dockstore-webservice/src/main/java/io/swagger/api/impl/ToolsApiServiceImpl.java

```diff
@@ -393,7 +399,27 @@ public Response toolsIdVersionsVersionIdTypeDescriptorRelativePathGet(String typ
     public Response toolsIdVersionsVersionIdTypeTestsGet(String type, String id, String versionId, SecurityContext securityContext)
             throws NotFoundException {
         /** we do not have test data implemented */
-        return Response.status(Response.Status.NOT_FOUND).build();
+        if (type == null) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        SourceFile.FileType fileType = getFileType(type);
+        if (fileType == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        switch (fileType) {
+            case DOCKSTORE_CWL:
+                return getFileByToolVersionID(id, versionId, CWL_TEST_JSON, null, false);
```

Comment:
This should be a bad request as well if DOCKSTORE_WDL is a bad request. It looks like the valid types are the two test json below.